js_modules/dagit/packages/core/src/asset-graph/AssetGraphExplorer.tsx

```diff
@@ -211,21 +246,47 @@ const AssetGraphExplorerWithData: React.FC<
         {
           ...explorerPath,
           opNames: [nextOpsNameSelection],
-          opsQuery: nextOpsQuery,
-          pipelineName: clicked.jobName || explorerPath.pipelineName,
+          opsQuery: nodeIsInDisplayedGraph
+            ? explorerPath.opsQuery
+            : `${explorerPath.opsQuery},++"${token}"++`,
+          pipelineName: explorerPath.pipelineName,
         },
         'replace',
       );
     },
-    [explorerPath, onChangeExplorerPath, findAssetInWorkspace, lastSelectedNode, assetGraphData],
+    [
+      isGlobalGraph,
+      explorerPath,
+      onChangeExplorerPath,
+      findJobForAsset,
+      history,
+      lastSelectedNode,
+      assetGraphData,
```

Comment:
A deps list like this makes me wonder whether `useCallback` remains worthwhile for this function.